GLOSSARY.md

```diff
@@ -0,0 +1,44 @@
+# Parts of Class Names
+
+| Name        | Meaning                                                                                                                                                                                                        |
+| ----------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Abstract... | An incomplete class used for *dynamic* polymorphism. It has pure virtual methods that need to be implemented by subclasses.                                                                                    |
+| Base...     | An incomplete class used for *static* polymorphism. It implements methods shared by all implementations. Mostly, it is used so that we have a non-templated superclass, which can be stored in e.g., a vector. |
+| ...Node     | An entry in a Logical Query Plan, for example a Predicate or a Join.                                                                                                                                           |
+| ...Operator | An entry in a Physical Query Plan, for example a Table Scan or a Sort-Merge Join. Currently, each operator is the physical representation of exactly one node. This is not a hard limitation.                  |
+
+# Approved Abbreviations
+
+We limit the use of abbreviations to very few cases. Unless an abbreviation is listed here or is something that we can expect every Computer Science Graduate to know (e.g., CSV, NUMA), do not use it in interfaces, code, or documentation.
+
+| Name        | Short for                                                              |
+| ----------- | ---------------------------------------------------------------------- |
+| ART         | Abstract Radix Tree (only for internal use of that data structure)     |
```

Comment:
Do we have an ART (Abstract Radix Tree)? Because we also have an ART (Adaptive Radix Tree) which is one of our two indexes.

Reply:
That's what I meant...